### ISSUE TITLE
`use-natspec` requires documenting unnamed parameters, but Solidity forbids `@param` for unnamed parameters

### ISSUE THREAD

**Issue by pvandervos-circle** [protofire/solhint#742]
Solidity allows unnamed function parameters (commonly used for intentional unused parameters and ABI/interface compatibility):

```solidity
function f(bytes calldata /*message*/) external {}
```

However, with Solhint’s `use-natspec` rule enabled, Solhint warns that the function has undocumented parameters:

```
Mismatch in @param count for function 'f'. Expected: 1, Found: 0
```

Trying to satisfy Solhint by adding an `@param` entry causes the Solidity compiler to fail, because the parameter has no name in the function signature:

```
Documented parameter "message" not found in the parameter list of the function.
```

This creates an incompatibility between Solhint and Solidity: the only ways to resolve are to (1) name the parameter in the function, or (2) disable `use-natspec` for that function/file.

---

#### Case 1: Valid Solidity, but Solhint warns

```solidity
pragma solidity ^0.8.30;

contract Test {
    /**
     * @notice Example
     */
    function f(bytes calldata /*message*/) external {}
}
```

Solhint:

```
Mismatch in @param count for function 'f'. Expected: 1, Found: 0
```

#### Case 2: Attempt to fix Solhint, but Solidity compiler fails

```solidity
pragma solidity ^0.8.30;

contract Test {
    /**
     * @notice Example
     * @param message Unused message payload
     */
    function f(bytes calldata /*message*/) external {}
}
```

Solidity compiler error:

```
Documented parameter "message" not found in the parameter list of the function.
```

---

### Expected Behavior

For unnamed parameters, Solhint should not require `@param` documentation, since Solidity cannot accept it.
